src/engine/execute.js

```diff
@@ -370,6 +382,19 @@ class BlockCached {
     }
 }
 
+const prepareBlockProfiling = function (profiler, blockCached) {
```

Comment:
We should probably add jsdoc to this function, and probably also follow the `_` naming convention, implying this is for private use.

Reply:
Added.